### ISSUE TITLE
[ENH] Move collection variance check out of _check_X so it only validates top-level estimator inputs

### ISSUE THREAD

**Issue by TonyBagnall** [aeon-toolkit/aeon#3599]
### Describe the feature or idea you want to propose

## Describe the issue

#3315 added `check_collection_variance()` to reject collections with near-zero but non-constant variance (`std <= 1e-7`, `range != 0`), because some numba utilities treat such series as effectively constant. It was wired into `BaseCollectionEstimator._check_X`, which runs on every call to a collection estimator's public API, including internal calls made by composite estimators on derived data.

This caused #3570: HIVE-COTE v2 crashed intermittently in the `hybrid.ipynb` notebook. The trigger is DrCIF's first-order-difference representation of `ItalyPowerDemand`: case 15 has a locally linear stretch, so `np.diff(X[15, 0])[13:16]` is constant up to float rounding (`std ≈ 5.3e-9`, `range ≈ 1.3e-8`). Whenever `RandomIntervals` drew that interval and passed the slice to `Catch22.fit_transform`, the variance check raised and aborted the whole HC2 fit.

Near-constant subseries are normal for interval, shapelet, and dictionary pipelines, and the error's advice ("rescale or normalise your data") is not appropriate: `ItalyPowerDemand` is already z-normalised.

As an interim fix, PR #3598 demoted the check in `_check_X` to a `UserWarning`. That stops the crashes but is not the final solution: users can now get spurious low-variance warnings caused by internal slices of valid data

For a test, `np.diff` of `ItalyPowerDemand` case 15, columns 13:16, is a real-world near-constant slice. A seeded `DrCIFClassifier` fit on `ItalyPowerDemand` covering that interval would catch any reintroduction.

Adding two data points from the HC2 speedup work. First, the cost side: DrCIF makes over a thousand validated Catch22 calls per fit/predict cycle (per tree, per representation), each paying the full _check_X  
  ▎ including the per-case std scan of check_collection_variance — validation inside the hot loop of the dominant HC2 component. Second, option 2 now has a precedent: the rocket transformers expose a private      
  ▎ kernels-only entry point (_transform_kernels) that Arsenal calls after validating once at ensemble level, keeping public calls fully checked and stored pipelines self-contained. The same pattern would fit the 
  ▎ interval forests' internal Catch22 usage, alongside restoring the top-level check to an error per option 1.                                                                                                      
                                                                                                                  

## Related

* #3315
* #3570
* PR #3598 , interim warning fix


### Describe your proposed solution


The check should distinguish top-level user input from internal component calls. Since internal components currently use the same public `fit`/`fit_transform` API as users, this needs one of the following:

1. Move the check up into the entry-point validation of `BaseClassifier`, `BaseRegressor`, `BaseClusterer`, and `BaseCollectionTransformer`, for example `_fit_setup`, out of `_check_X`, so it runs once on the data the user actually passed.

2. Have composite estimators call component transformers through a private path that skips input revalidation. This would also avoid redundant conversion and metadata work on every slice, which would be a performance win for interval forests.

3. Gate the check behind a tag or constructor flag that composite estimators disable on their internal components.

Option 1 seems the smallest coherent change. Option 2 is more invasive but fixes the underlying layering problem. Whichever is chosen, the warning added in PR #3598  should then be restored to an error at the top level only.


### Describe alternatives you've considered, if relevant

_No response_

### Additional context

_No response_

**Comment by TonyBagnall** [comment]
if we do 2) can we just call _fit/_predict/_fit_predict?

**Comment by MatthewMiddlehurst** [comment]
Would be good to investigate whether we can do that, because otherwise it will be spamming warnings.

**Comment by TonyBagnall** [comment]
its already spamming warnings, which is however better than raising errors, 50+ warnings per DrCIF. But this does identify another inefficiency with DrCIF, our slowest model in the new world order. It is calling full data validation through Catch22 on every slice, meaning it is doing it 1000s of time per call. 

```python
RandomIntervals (aeon/transformations/collection/interval_based/_random_intervals.py):                                                                                  
                                                                                                                                                                                                                     
  Fit path — line 372, inside _generate_interval, in the for feature in self._features: loop (line 364):                                                                                                             
  t = feature.fit_transform(                                                                                                                                                                                         
      np.expand_dims(                                                                                                                                                                                                
          X[:, dim, interval_start:interval_end:dilation], axis=1                                                                                                                                                    
      ),                                                                                                                                                                                                             
      y,                                                                                                                                                                                                             
  )                                                             
```
 Predict path — line 418, inside _transform_interval:                                                                                                                                                               
```python
  Xt = feature.transform(                                                                                                                                                                                            
      np.expand_dims(X[:, dim, interval_start:interval_end:dilation], axis=1)                                                                                                                                        
  )                                                                                                                                                                                                                  
```
this expands to a large number of calls through 
1. RandomIntervals._fit_transform (line 160) / _transform (line 245) call _generate_interval / _transform_interval once per interval self.n_intervals times.                                       
 2. Inside each of those, the for feature in self._features loop executes line 372/418 once per feature.                                                                                                                
                                                                                                                                                                                                                     
  So one RandomIntervals call = n_intervals × n_features slice-validations. And RandomIntervals itself is invoked once per tree per representation by the forest — intervals[r].fit_transform(...) at                
  base_interval_forest.py:1012 (fit) and intervals[r].transform(...) at :1086 (predict). Total ≈ n_trees × n_representations × n_intervals × n_features, which is the "thousand-plus" figure in the issue. 

Quick mock up by claude says 
Measured on a 60×1×200 dataset, estimating 5000 calls, with a 5%-10% overhead from redundant validation. The solution for me is to simply bypass all these redundant calls and add a little more speed to DrCIF, which is now 80% of HC2 run time.

**Comment by TonyBagnall** [comment]
so Im hoping we can completely resolve this by the simple expedient of calling _fit_transform and _transform on those two hot lines. It works for Catch22, but need it to work for everything. In my mind the _transform contract is "dont do any validation", but need to check.

Annoyingly supervised intervals internally converts to 2D and passes 2D slices around,

**Comment by TonyBagnall** [comment]
so solution is to make SupervisedIntervals behave correctly as Catch22 does, then we get a 10% speed up of DrCIF for free!
